src/op.cpp

```diff
@@ -2639,9 +2650,29 @@ uint32_t jitc_var_scatter_packet(size_t n, uint32_t target_,
                              mode == ReduceMode::NoConflicts ||
                              (mode == ReduceMode::Auto &&
                               target_info.size <= llvm_expand_threshold));
+        } else if (backend == JitBackend::CUDA) {
+            use_packet_op = (mode == ReduceMode::Expand ||
+                             mode == ReduceMode::Permute ||
+                             mode == ReduceMode::NoConflicts ||
+                             mode == ReduceMode::Auto);
+
+            VarType targe_type = (VarType) jitc_var(target)->type;
+            if (compute_capability < 90)
+                use_packet_op = use_packet_op && op == ReduceOp::Add &&
+                                targe_type == VarType::Float16;
+            else
+                use_packet_op =
+                    use_packet_op &&
+                    ((targe_type == VarType::Float32 && op == ReduceOp::Add) ||
+                     (targe_type == VarType::Float16 &&
+                      (op == ReduceOp::Add || op == ReduceOp::Min ||
+                       op == ReduceOp::Max)));
         }
     }
 
+    // If the packet size is not divisible by two we cannot use packet ops.
+    use_packet_op = use_packet_op && n > 1 && n % 2 == 0;
```

Comment:
If we move the logic to code generation, perhaps it's easier to not even have special handling for the n=1 case here.